cli_tools/common/utils/storage/storage_client.go

```diff
@@ -172,6 +184,27 @@ func (sc *Client) FindGcsFile(gcsDirectoryPath string, fileExtension string) (*s
 		"path %v doesn't contain a file with %v extension", gcsDirectoryPath, fileExtension)
 }
 
+func isDepthValid(lookupDepth int, lookupPath, objectPath string) bool {
```

Comment:
This is a bit confusion to me. It looks not like a recursive. Will the it.next() cover all files in any depth of subfolders?

Reply:
Recursive as in going through the full GCS folder system instead of checking only the current level. The recursive part doesn't happen in our code but instead in the results of GCS lookup.

Reply:
I had the same instinct as Tony (I clarified offline with Zoran), so maybe it's worth adjusting the wording?

Reply:
@zoran15 if that is the case, it seems that depth check can't reduce the lookup workload. Every single files will be gone through. Per my understanding, depth check is usually for limiting the workload. Here it doesn't help.

Reply:
Updated the description of this PR so "recursive" is not mentioned.

@dntczdx You're right that the lookup code will still look at all the files within a search path. However, this fix is not about reducing the workload or the number of files looked at. It's about correctness: customers provide GCS path to a folder that should contain OVF archive including a descriptor. This lookup should not try to search for OVF archives in any of the subfolders, but only at the level provided by the customer (the flag is currently behaving like a prefix search which is not correct). E.g. if a customer provides "gs://bucket/" as gcs path, we will currently check all subfolders and pick the first that contains an .ovf descriptor. This can lead to unintended OVF archives from being imported.